src/module/rolls/roll.ts

```diff
@@ -0,0 +1,57 @@
+// TODO:
+// - Convert to HTML for chat messages
+// - Rehydrate from chat message (will be needed for rerolls)
+// - Options input covers all cases
+// - Rolling that plays nicer with DF Manual Rolls (all in one go, not {d6+N,d10,d10})
+// - Rerolls update chat message
+
+export enum HIT_TYPE {
+  MISS = 'MISS',
+  WEAK = 'WEAK',
+  STRONG = 'STRONG',
+}
+
+export interface PreRollOptions {
+  automaticOutcome?: HIT_TYPE
+  presetActionDie?: number // As in Armored #1
+  threeChallengeDice?: boolean // As in Sleuth #1
+}
+
+export interface PostRollOptions {
+  actionBonus?: number // As in Kinetic #2
```

Comment:
i'd recommend calling it "add" or "adds" to keep it consistent with the game terminology. ditto the use of "bonus" elsewhere in the module, particularly in localizable labels (where inconsistency with the original text can make life more difficult for people doing the localization).